Meta/Lagom/.gitignore

```diff
@@ -1,5 +1,5 @@
-liblagom.a
 CMakeCache.txt
 CMakeFiles
-Makefile
+build
```

Comment:
Is that not covered by the top-level `.gitignore`'s `build`/`Build` already?

Reply:
Are `.gitignore` rules applied recursively? i.e. does `build` in toplevel .gitignore ignore ever/path/build?

Reply:
yes :)

Reply:
I don't understand the state of this PR, do you intend to make changes following this comment @nico?

Reply:
Ping

Reply:
State is that I need to apparently read some docs on `.gitignore`. Like, if that's how it works, why do we have `Toolchain/Build` in toplevel .gitignore at all? Isn't that already redundant?

Reply:
Yup, that was redundant :)